src/Compilers/CSharp/Test/Emit3/RefStructInterfacesTests.cs

```diff
@@ -17631,5 +17744,420 @@ public void RefFieldTypeAllowsRefLike()
+{
+    public delegate void Action<T1, T2>(T1 x, T2 y) where T1 : allows ref struct where T2 : allows ref struct;
+}
+";
+            var comp = CreateCompilation(src, targetFramework: s_targetFrameworkSupportingByRefLikeGenerics, options: TestOptions.UnsafeReleaseExe);
+
+            CompileAndVerify(
+                comp, expectedOutput: ExecutionConditionUtil.IsMonoOrCoreClr ? @"<>f__AnonymousDelegate0`1[S1] Test1 S1" : null,
+                verify: ExecutionConditionUtil.IsMonoOrCoreClr ? Verification.Passes : Verification.Skipped,
+                symbolValidator: (m) =>
+                {
+                    foreach (var tp in m.ContainingAssembly.GetTypeByMetadataName("<>f__AnonymousDelegate0`1").TypeParameters)
+                    {
+                        Assert.True(tp.AllowByRefLike);
+                    }
+                }
+                ).VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void AnonymousDelegateType_06_PartiallyGenericAnonymousDelegate_CannotAllowRefStruct()
+        {
+            var src = @"
+class C
+{
+    static void Main()
+    {
+        Test1(new S1());
+    }
+    
+    static void Test1<T>(T x)
+    {
+        var d = Helper<T>.Test1;
+        System.Console.Write(d.GetType());
+        System.Console.Write("" "");
+        d(x, new S2());
+    }
+}
+
+class Helper<T>
+{
+    public static void Test1(T x, S2 y)
+    {
+        System.Console.Write(""Test1"");
+        System.Console.Write("" "");
+        System.Console.Write(typeof(T));
+    }
+}
+
+struct S1 {}
+
+ref struct S2 {}
+";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net70, options: TestOptions.ReleaseExe);
+
+            CompileAndVerify(
+                comp, expectedOutput: ExecutionConditionUtil.IsMonoOrCoreClr ? @"<>f__AnonymousDelegate0`1[S1] Test1 S1" : null,
+                verify: ExecutionConditionUtil.IsMonoOrCoreClr ? Verification.Passes : Verification.Skipped,
+                symbolValidator: (m) =>
+                {
+                    foreach (var tp in m.ContainingAssembly.GetTypeByMetadataName("<>f__AnonymousDelegate0`1").TypeParameters)
+                    {
+                        Assert.False(tp.AllowByRefLike);
+                    }
+                }
+                ).VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void AnonymousDelegateType_07_ActionDisallowsRefStruct_CannotAllowRefStruct()
+        {
+            var src = @"
+class C
+{
+    static void Main()
+    {
+        Test1(new S1());
+    }
+    
+    static void Test1(S1 x)
+    {
+        var d = Helper.Test1;
+        System.Console.Write(d.GetType());
+        System.Console.Write("" "");
+        d(x, new S2());
+    }
+}
+
+class Helper
+{
+    public static void Test1(S1 x, S2 y)
+    {
+        System.Console.Write(""Test1"");
+    }
+}
+
+ref struct S1 {}
+
+ref struct S2 {}
+
+namespace System
+{
+    public delegate void Action<T1, T2>(T1 x, T2 y);
+}
```

Comment:
Do we need both this definition and `TargetFramework.Net70`?

Reply:
> Do we need both this definition and TargetFramework.Net70? Same question for the next test.

I prefer to explicitly set up the desired test environment.